src/deepsparse/clip/pipelines.py

```diff
@@ -0,0 +1,249 @@
+
+
+@Pipeline.register(task="clip_text", default_model_path=None)
+class CLIPTextPipeline(Pipeline):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        self.tokenizer = tokenize
+        # TODO: This is a placeholder. The onnx model should apply the
+        # text_projection in the forward pass but due to a current argmax export error,
+        # including the text_projection in the post-processing and using dummy values.
+        self.text_projection = np.random.rand(512, 512)
+
+    @property
+    def input_schema(self) -> Type[CLIPTextInput]:
+        """
+        :return: pydantic model class that inputs to this pipeline must comply to
+        """
+        return CLIPTextInput
+
+    @property
+    def output_schema(self) -> Type[CLIPTextOutput]:
+        """
+        :return: pydantic model class that inputs to this pipeline must comply to
+        """
+        return CLIPTextOutput
+
+    def setup_onnx_file_path(self):
+        """
+        Performs any setup to unwrap and process the given `model_path` and other
+        class properties into an inference ready onnx file to be compiled by the
+        engine of the pipeline
+
+        :return: file path to the ONNX file for the engine to compile
+        """
+        return model_to_path(self.model_path)
+
+    def process_inputs(
+        self, inputs: CLIPTextInput
+    ) -> Tuple[List[np.ndarray], Dict[str, Any]]:
+        """
+        Preprocess inputs for CLIP's Trext Branch to comply with the DeepSparse Engine
+
+        :param inputs: CLITextInput
+        :return: list of preprocessed numpy arrays
+        """
+        if isinstance(inputs.text, str):
+            inputs.text = [inputs.text]
+
+        tokens = self.tokenizer(inputs.text)
+        max_dim = np.array(tokens.argmax(dim=-1))
+        tokens = [np.array(t) for t in tokens]
+        tokens = np.stack(tokens, axis=0)
+        return [tokens], dict(max_dim=max_dim)
+
+    def process_engine_outputs(
+        self, engine_outputs: List[np.array], **kwargs
+    ) -> CLIPTextOutput:
+        """
+        :param engine_outputs: list of numpy arrays that are the output of the engine
+            forward pass
+        :return: outputs of engine post-processed into an object in the `output_schema`
+            format of this pipeline
+        """
+        max_dim = kwargs["max_dim"]
+        engine_outputs = engine_outputs[0]
+        # NOTE: This operation should be part of the foward pass. Moved here due to
+        # onnx export error
+        embeddings = (
+            engine_outputs[np.arange(engine_outputs.shape[0]), max_dim]
+            @ self.text_projection
+        )
+        return self.output_schema(text_embeddings=[embeddings])
+
+
+@BasePipeline.register(task="clip_zeroshot", default_model_path=None)
+class CLIPZeroShotPipeline(BasePipeline):
+    def __init__(self, visual_args: Dict, text_args: Dict, **kwargs):
```

Comment:
let's explicitly name visual args and text args here so user doesn't need to dig into the underlying pipelines